AKS release notes v20250617 (#5091)
Azure/AKS: CHANGELOG.md

Azure Kubernetes Service will no longer support the --skip-gpu-driver-install node pool tag to skip automatic driver installation. Starting on August 14 2025, you will no longer be able to use this node pool tag at AKS node pool creation time to install custom GPU drivers or use the GPU Operator. Alternatively, you should use the generally available gpu-driver API field to update your existing node pools or create new GPU-enabled node pools to [skip automatic GPU driver installation](https://learn.microsoft.com/azure/aks/gpu-cluster?tabs=add-ubuntu-gpu-node-pool#skip-gpu-driver-installation-preview).

> 

Reviewer: 1.27 LTS version and 1.30 community version are going out of support by July 30th. Please upgrade to a supported version , refer AKS calendar https://learn.microsoft.com/azure/aks/supported-kubernetes-versions?tabs=azure-cli#aks-kubernetes-release-calendar

Reviewer: 1.33 Community version is going to be released before end of June 2025, please keep track of the rollout by regions via Release tracker.

### Release Notes

* Preview Features
  * [Azure Monitor Application Insights for Azure Kubernetes Service (AKS) workloads](https://learn.microsoft.com/azure/azure-monitor/app/kubernetes-codeless) is now available in preview.
  
* Features  
  * AKS patch versions 1.32.5, 1.31.9 are now available. Refer to [version support policy](https://learn.microsoft.com/azure/aks/supported-kubernetes-versions?tabs=azure-cli#kubernetes-version-support-policy) and [upgrading a cluster](https://learn.microsoft.com/azure/aks/upgrade-aks-cluster?tabs=azure-cli) for more information.

>   * AKS now allows [daily schedules](https://learn.microsoft.com/azure/aks/planned-maintenance?tabs=azure-cli#schedule-types) for the auto upgrade configuration.

Reviewer: This is a behavior change and not a feature.  Also the patch releases should likely be called under announcement pls move it.

Author: > This is a behavior change and not a feature. Also the patch releases should likely be called under announcement pls move it.

All patch releases were put under "features" section in previous release notes.

>   * API Server VNet Integration is available now, please find the [most up to date regions](https://learn.microsoft.com/en-us/azure/aks/api-server-vnet-integration#limited-availability) where this feature has been rolled out.

Reviewer: Remove en-us from link

  
* Bug Fixes

>   * Fixed a race condition with streams sharing data between Cilium agent and ACNS security agent.

Reviewer: Do we have a GH link or doc on the issue for better clarity.

Author: > Do we have a GH link or doc on the issue for better clarity.

Checking with Sam Foo about the GH issue link.

Author: It turns out this issue was reported thru CRI/IcMs.

>   * Fixed an issue by forcing a pinned driver file name for NVv3 VMs, ensuring they do not use unsupported GRID driver versions.

Reviewer 2: What was the issue, not the fix?

>   * Fixed Azure Policy addon Gatekeeper regression causing crash loop on clusters with Kubernetes versions < 1.27.

Reviewer: GH link to this ?

>   * Resolved an issue where node pool scaling failed when customized kubelet configuration applied. Previously, node pools using CustomKubeletConfigs could not be scaled, and encountered an error message stating that the CustomKubeletConfig or CustomLinuxOSConfig cannot be changed for the scaling operation.

Reviewer 2: ```suggestion
  * Resolved an issue where node pool scaling failed with customized kubelet configuration. Previously, node pools using CustomKubeletConfigs could not be scaled, and encountered an error message stating that the CustomKubeletConfig or CustomLinuxOSConfig cannot be changed for the scaling operation.
```

Reviewer 2: What was the issue? Was it what is stated after "Previously"?

>   * Resolved an issue where node pool scaling failed when customized kubelet configuration applied. Previously, node pools using CustomKubeletConfigs could not be scaled, and encountered an error message stating that the CustomKubeletConfig or CustomLinuxOSConfig cannot be changed for the scaling operation.

Reviewer 2: ```suggestion
  * Resolved an issue where node pool scaling failed with customized kubelet configuration. Previously, node pools using CustomKubeletConfigs could not be scaled, and encountered an error message stating that the CustomKubeletConfig or CustomLinuxOSConfig cannot be changed for the scaling operation.
```

>   * Resolved an issue where node pool scaling failed when customized kubelet configuration applied. Previously, node pools using CustomKubeletConfigs could not be scaled, and encountered an error message stating that the CustomKubeletConfig or CustomLinuxOSConfig cannot be changed for the scaling operation.

Reviewer: Any links again.

Author: > Any links again.

It is fixing a CRI - https://portal.microsofticm.com/imp/v5/incidents/details/625203368/summary

  * Fixed an [issue](https://github.com/kubernetes-sigs/cloud-provider-azure/issues/8649) where updating node pools with the exclude label, it doesn't update the Load Balancer Backend Pool properly.

>   * Resolved a problem when upgrading Kubenet or Nodesubnet cluster with AGIC enabled to Azure CNI Overlay there might be some connectivity issues to services exposed via Ingress App Gateway public IP.

Reviewer: Again all these errors and fixes are fine but with a GH link or doc it becomes more easy for CX to relate and understand fully.

Author: > Again all these errors and fixes are fine but with a GH link or doc it becomes more easy for CX to relate and understand fully.

@chasewilson do you have the GitHub issue for this fix (https://dev.azure.com/msazure/CloudNativeCompute/_git/aks-rp/pullrequest/12622372), or is this issue reported by a CRI?

Author: Checked with Roma Bogatikov, this was something that we noticed internally, and we don't have any GitHub issues or ICMs for this.

  * Fix a bug where clusters with Node Auto Provisioning enabled could intermittently get an error about "multiple identities configured" and be unable to authenticate with Azure.

>   * Fixed an issure to ensure the vms in a specific cloud are compatible with the latest Windows 550 grid driver.

Reviewer 2: ```suggestion
  * Fixed an issue to ensure the virtual machines in a specific cloud are compatible with the latest Windows 550 grid driver.
```
Unclear what the issue was. What specific cloud? What driver?

Author: > Unclear what the issue was. What specific cloud? What driver?

@sachi-desai, could you please share the issue addressed by [this PR](https://dev.azure.com/msazure/CloudNativeCompute/_git/aks-rp/pullrequest/12619047)?

>   * Fixed a bug for the policy addon by addressing a sync issue with allowed service accounts.

Reviewer 2: What was the issue, not the fix?

Author: > What was the issue, not the fix?

@qpetraroia could you please share what was the issue addressed by [this PR](https://dev.azure.com/msazure/CloudNativeCompute/_git/aks-rp/pullrequest/12676609)?

  

* Behavior Changes  

>   * Static egress gateway memory limits increased from 500Mi to 3000Mi reducing the risk of memory-related restarts under load.

Reviewer: Link for this either on GH or doc calling out this change and benefit.

>   * GPU-Provisioner is a component of KAITO, and it is now hidden from the customer as part of the KAITO add-on.

Reviewer: Link

  * [Azure Monitor managed service for Prometheus](https://learn.microsoft.com/azure/azure-monitor/essentials/prometheus-metrics-overview) updates the max shards from 12 to 24, ensuring enhanced scaling capabilities.

>   * `linuxutil plugin` is enabled again for [Retina Basic and ACNS](https://retina.sh/docs/Metrics/plugins/Linux/linuxutil).

Reviewer: Component update this one.

>   * Kubelet Service Certificate Rotation has now been rolled out to East US and UK South. Existing node pools will have kubelet serving certificate rotation enabled by default when they perform their first upgrade to any kubernetes version 1.27 or greater. New node pools on kubernetes version 1.27 or greater will have kubelet serving certificate rotation enabled by default. For more information on kubelet serving certificate rotation and disablement, see [certificate rotation in Azure Kubernetes Service](https://learn.microsoft.com/azure/aks/certificate-rotation).

Reviewer: This probably should be under features.

Author: > This probably should be under features.

@allyford do you think this KSCR should be a feature or behavior change?

Reviewer 3: Feature please

Author: Moved.

>   * Node Auto-Provisioning (NAP) now requires Kubernetes RBAC to be enabled also. For more info, see [RBAC for Kubernetes](https://learn.microsoft.com/azure/aks/manage-azure-rbac).

Reviewer 2: Documentation or explanation?

>   * Safeguards now supports policy assignments through its own first-party application.

Reviewer 2: A bit of an implementation detail. Needs documentation link on what permissions are now needed or what changed.

Author: > A bit of an implementation detail. Needs documentation link on what permissions are now needed or what changed.

@qpetraroia do you have document link or more details on what changed?

Reviewer 4: @AllenWen-at-Azure  - you can take out this safeguards line

> With this enhancement, Safeguards no longer relies on OBO tokens to assign policies. As a result, customer permission validation during policy assignment is no longer required, simplifying the experience and reducing friction.

Reviewer 2: "OBO tokens" isn't a concept that's user friendly.

Reviewer 4: @AllenWen-at-Azure -

"Deployment Safeguards no longer requires Azure Policy permissions. Cluster admins will have the ability to turn on and disable Deployment Safeguards."

* Component Updates  
  * Windows node images  
    * Server 2019 Gen1 – [`17763.7314.250518`](vhd-notes/AKSWindows/2019/17763.7314.250518.txt).
    * Server 2022 Gen1/Gen2 – [`20348.3692.250518`](vhd-notes/AKSWindows/2022/20348.3692.250518.txt)
    * Server 23H2 Gen1/Gen2 – [`25398.1611.250518`](vhd-notes/AKSWindows/23H2/25398.1611.250518.txt)
  * AKS Azure Linux v2 image has been updated to [202506.12.0](vhd-notes/AzureLinux/202506.12.0.txt).
  * AKS Azure Linux v3 image has been updated to [202506.12.0](vhd-notes/AzureLinuxv3/202506.12.0.txt).
  * AKS Ubuntu 22.04 node image has been updated to [202506.12.0](vhd-notes/aks-ubuntu/AKSUbuntu-2204/202506.12.0.txt).
  * AKS Ubuntu 24.04 node image has been updated to [202506.12.0](vhd-notes/aks-ubuntu/AKSUbuntu-2404/202506.12.0.txt).
  * Updated Istio-based service mesh add-on to versions [v1.23.6](https://istio.io/latest/news/releases/1.23.x/announcing-1.23.6/), [v1.24.6](https://istio.io/latest/news/releases/1.24.x/announcing-1.24.6/), [v1.25.3](https://istio.io/latest/news/releases/1.25.x/announcing-1.25.3/).

>   * Updated secret store driver to [v1.5.0](https://github.com/kubernetes-sigs/secrets-store-csi-driver/releases/tag/v1.5.0) and azure provider to [v1.7.0](https://github.com/Azure/secrets-store-csi-driver-provider-azure/releases/tag/v1.7.0).

Reviewer 2: ```suggestion
  * Updated Secret Store CSI driver to [v1.5.0](https://github.com/kubernetes-sigs/secrets-store-csi-driver/releases/tag/v1.5.0) and the Azure provider to [v1.7.0](https://github.com/Azure/secrets-store-csi-driver-provider-azure/releases/tag/v1.7.0).
```

  * Updated Azure Monitor managed service for Prometheus images to [05-29-2025 release](https://github.com/Azure/prometheus-collector/blob/main/RELEASENOTES.md#release-05-29-2025).
  * Updated Azure Policy add-on image to [v1.12.2](https://learn.microsoft.com/azure/governance/policy/concepts/policy-for-kubernetes#1122).
  * Updated Retina basic image to [v0.0.34](https://github.com/microsoft/retina/releases/tag/v0.0.34).
  * Updated Retina Enteprise image to [v0.1.10](https://github.com/azure-networking/retina-enterprise/releases/tag/v0.1.10).
  * Updated Metrics-server to [v0.7.2-7](https://github.com/aks-lts/metrics-server/releases/tag/v0.7.2-7) for AKS clusters on 1.32+ version, [v0.6.3-6](https://github.com/aks-lts/metrics-server/releases/tag/v0.6.3-6) for AKS clusters on 1.24+ version.
  * Updated coredns version to [v1.9.4-6](https://github.com/aks-lts/coredns/releases/tag/v1.9.4-6) for AKS clusters on 1.24+ version, [v1.11.3-8](https://github.com/aks-lts/coredns/releases/tag/v1.11.3-8) for AKS clusters on 1.32+ version, [v1.12.1-2](https://github.com/aks-lts/coredns/releases/tag/v1.12.1-2) for AKS clusters on 1.33+ version.

>   * Updated cilium/fqdn images to [v1.14.20](https://github.com/azure-networking/cilium-private/releases/tag/core%2Fv1.14.20-1) for Kubernetes 1.29, [v1.16.10](https://github.com/azure-networking/cilium-private/releases/tag/core%2Fv1.16.10) for Kubernetes 1.31, [v1.17.4](https://github.com/azure-networking/cilium-private/releases/tag/core%2Fv1.17.4) for Kubernetes 1.32.

Reviewer 2: ```suggestion
  * Updated cilium/fqdn images to [v1.14.20](https://github.com/azure-networking/cilium-private/releases/tag/core%2Fv1.14.20-1) for Kubernetes 1.29, [v1.16.10](https://github.com/azure-networking/cilium-private/releases/tag/core%2Fv1.16.10) for Kubernetes 1.31, [v1.17.4](https://github.com/azure-networking/cilium-private/releases/tag/core%2Fv1.17.4) for Kubernetes 1.32.
```
What are these Cilium private repo links?

Author: > What are these Cilium private repo links?

Changed to OSS releases link for v1.16/v1.17, but there is no linkage for v1.14.20, checking with the PR owner.

## Release 2025-06-17

Monitor the release status by region at [AKS-Release-Tracker](https://releases.aks.azure.com/). This release is titled `v20250617`.

### Announcements

* Kubernetes 1.27 LTS version and 1.30 community version are going out of support by July 30th. Please upgrade to a supported version , refer to [AKS release calendar](https://learn.microsoft.com/azure/aks/supported-kubernetes-versions?tabs=azure-cli#aks-kubernetes-release-calendar) for more information.
* Kubernetes 1.33 community version is going to be released before end of June 2025, please keep track of the rollout by regions via [AKS Release tracker](https://releases.aks.azure.com/).
* Customers using Azure Linux 2.0 should migrate to Azure Linux 3.0 before November 2025. For details on how to migrate from Azure Linux 2.0 to Azure Linux 3.0, see this [doc](https://learn.microsoft.com/azure/azure-linux/how-to-enable-azure-linux-3). AKS is currently working on a feature to allow for migrations between Azure Linux 2.0 and Azure Linux 3.0 through a node pool update command. For updates on feature progress and availability, see Github [issue](https://github.com/Azure/AKS/issues/4987).

> * Starting in June 2025, AKS clusters with version >= 1.28 and using Azure Linux 2.0 can be opted into [Long Term Support](https://learn.microsoft.com/azure/aks/long-term-support).

Reviewer 3: ```suggestion
* Starting in June 2025, AKS clusters with version >= 1.28 and using Azure Linux 2.0 can be opted into [Long Term Support](https://learn.microsoft.com/azure/aks/long-term-support). See [blog](https://techcommunity.microsoft.com/blog/linuxandopensourceblog/azure-linux-now-supports-aks-long-term-support-lts-starting-with-kubernetes-v1-2/4424826) for more information.
```

> * Starting on 17 June 2025, AKS will no longer create new node images for [Ubuntu 18.04](https://github.com/Azure/AKS/issues/4873) or provide security updates. Existing node images will be deleted. Your node pools will be unsupported and you will no longer be able to scale. To avoid service disruptions, scaling restrictions, and remain supported, please follow our instructions to [upgrade](https://learn.microsoft.com/azure/aks/upgrade-aks-cluster?tabs=azure-cli) to a [supported Kubernetes version](https://learn.microsoft.com/azure/aks/supported-kubernetes-versions).

Reviewer 3: ```suggestion
* [Ubuntu 18.04](https://github.com/Azure/AKS/issues/4873) is no longer supported on AKS. AKS will no longer create new node images or provide security updates for Ubuntu 18.04 nodes. Existing node images will be deleted by 17 July 2025. Scaling operations will fail. To avoid service disruptions, scaling restrictions, and remain supported, please follow our instructions to [upgrade](https://learn.microsoft.com/azure/aks/upgrade-aks-cluster?tabs=azure-cli) to a [supported Kubernetes version](https://learn.microsoft.com/azure/aks/supported-kubernetes-versions).
```

* [Teleport (preview)](https://github.com/Azure/acr/blob/main/docs/teleport/aks-getting-started.md) on AKS will be retired on 15 July 2025, please [migrate to Artifact Streaming (preview) on AKS](https://learn.microsoft.com/azure/aks/artifact-streaming) or update your node pools to set --aks-custom-headers EnableACRTeleport=false. Azure Container Registry has removed the Teleport API meaning that any nodes with Teleport enabled are pulling images from Azure Container Registry as any other AKS node. After 15 July 2025, any node pools with Teleport (preview) enabled may experience breakage and node provisioning failures. For more information, see [aka.ms/aks/teleport-retirement](https://aka.ms/aks/teleport-retirement).

> Azure Kubernetes Service will no longer support the --skip-gpu-driver-install node pool tag to skip automatic driver installation. Starting on August 14 2025, you will no longer be able to use this node pool tag at AKS node pool creation time to install custom GPU drivers or use the GPU Operator. Alternatively, you should use the generally available gpu-driver API field to update your existing node pools or create new GPU-enabled node pools to [skip automatic GPU driver installation](https://learn.microsoft.com/azure/aks/gpu-cluster?tabs=add-ubuntu-gpu-node-pool#skip-gpu-driver-installation-preview).

Reviewer 5: Adjusted the GA docs link

### Release Notes

* Preview Features

>   * [Azure Monitor Application Insights for Azure Kubernetes Service (AKS) workloads](https://learn.microsoft.com/azure/azure-monitor/app/kubernetes-codeless) is now available in preview.

Reviewer 3: ```suggestion
  * [Azure Monitor Application Insights for Azure Kubernetes Service (AKS) workloads](https://learn.microsoft.com/azure/azure-monitor/app/kubernetes-codeless) is now available in preview.
    * Ubuntu 24.04 is now available in public preview in k8s 1.32+. ContainerD 2.0 is enabled by default. You can create new Ubuntu 24.04 node pools or update existing Linux node pools to Ubuntu 24.04. Use the "Ubuntu2404" os sku enum after registering the preview flag "Ubuntu2404Preview". You can also use the new "Ubuntu2204" os sku enum to rollback to Ubuntu 22.04 if you encounter any issues! You may also rollback using "Ubuntu" os sku enum. For more information on upgrading your OS version, see https://aka.ms/aks/upgrade-os-version.
```

  * Fixed an [issue](https://github.com/kubernetes-sigs/cloud-provider-azure/issues/8649) where updating node pools with the exclude label, it doesn't update the Load Balancer Backend Pool properly.
  * Resolved a problem when upgrading Kubenet or Nodesubnet cluster with AGIC enabled to Azure CNI Overlay there might be some connectivity issues to services exposed via Ingress App Gateway public IP.
  * Fix a bug where clusters with Node Auto Provisioning enabled could intermittently get an error about "multiple identities configured" and be unable to authenticate with Azure.
  * Fixed an issure to ensure the vms in a specific cloud are compatible with the latest Windows 550 grid driver.
  * Fixed a bug for the policy addon by addressing a sync issue with allowed service accounts.
  

* Behavior Changes  
  * AKS now allows [daily schedules](https://learn.microsoft.com/azure/aks/planned-maintenance?tabs=azure-cli#schedule-types) for the auto upgrade configuration.

>   * [Static Egress Gateway](https://learn.microsoft.com/azure/aks/configure-static-egress-gateway) memory limits increased from 500Mi to 3000Mi reducing the risk of memory-related restarts under load.

Reviewer 3: ```suggestion
  * [Static Egress Gateway](https://learn.microsoft.com/azure/aks/configure-static-egress-gateway) memory limits increased from 500Mi to 3000Mi reducing the risk of memory-related restarts under load.
    * [HTTP Proxy](https://aka.ms/aks/http-proxy) configuration updates will now trigger automatic reimaging with this release. This functionality was temporarily disabled.
```

  * GPU-Provisioner is a component of [KAITO](https://learn.microsoft.com/azure/aks/ai-toolchain-operator), and it is now hidden from the customer as part of the KAITO add-on.
  * [Azure Monitor managed service for Prometheus](https://learn.microsoft.com/azure/azure-monitor/essentials/prometheus-metrics-overview) updates the max shards from 12 to 24, ensuring enhanced scaling capabilities.
  * `linuxutil plugin` is enabled again for [Retina Basic and ACNS](https://retina.sh/docs/Metrics/plugins/Linux/linuxutil).
  * Kubelet Service Certificate Rotation has now been rolled out to East US and UK South. Existing node pools will have kubelet serving certificate rotation enabled by default when they perform their first upgrade to any kubernetes version 1.27 or greater. New node pools on kubernetes version 1.27 or greater will have kubelet serving certificate rotation enabled by default. For more information on kubelet serving certificate rotation and disablement, see [certificate rotation in Azure Kubernetes Service](https://learn.microsoft.com/azure/aks/certificate-rotation).

>   * Node Auto-Provisioning (NAP) now requires Kubernetes RBAC to be enabled, because NAP relies on secure and scoped access to Kubernetes resources to provision nodes based on pending pod resource requests. For more information, see [RBAC for Kubernetes](https://learn.microsoft.com/azure/aks/manage-azure-rbac).

Author: > It would be good to make it clear if it requires Kubernetes RBAC or Azure RBAC as the link is to Azure.
> 
> Is it just RBAC needs to be enabled, or Entra ID with Kubernetes RBAC, or Entra ID with Azure RBAC?

@wdarko1 could you please help clarify?

>   * Node Auto-Provisioning (NAP) now requires Kubernetes RBAC to be enabled, because NAP relies on secure and scoped access to Kubernetes resources to provision nodes based on pending pod resource requests. For more information, see [RBAC for Kubernetes](https://learn.microsoft.com/azure/aks/manage-azure-rbac).

Reviewer 6: NAP requires Kubernetes RBAC, which is enabled by default.

>   * Node Auto-Provisioning (NAP) now requires Kubernetes RBAC to be enabled, because NAP relies on secure and scoped access to Kubernetes resources to provision nodes based on pending pod resource requests. For more information, see [RBAC for Kubernetes](https://learn.microsoft.com/azure/aks/manage-azure-rbac).

Author: updated.

>   * Node Auto-Provisioning (NAP) now requires Kubernetes RBAC to be enabled, because NAP relies on secure and scoped access to Kubernetes resources to provision nodes based on pending pod resource requests. For more information, see [RBAC for Kubernetes](https://learn.microsoft.com/azure/aks/manage-azure-rbac).

Reviewer 6: ```suggestion
  * Node Auto-Provisioning (NAP) now requires Kubernetes RBAC to be enabled, because NAP relies on secure and scoped access to Kubernetes resources to provision nodes based on pending pod resource requests. Kubernetes RBAC is enabled by default. For more information, see [RBAC for Kubernetes](https://kubernetes.io/docs/reference/access-authn-authz/rbac/).
```

  * Resolved a problem when upgrading Kubenet or Nodesubnet cluster with AGIC enabled to Azure CNI Overlay there might be some connectivity issues to services exposed via Ingress App Gateway public IP.
  * Fixed a bug where clusters with Node Auto Provisioning enabled could intermittently get an error about "multiple identities configured" and be unable to authenticate with Azure.
  * Fixed an issure to ensure the vms in a specific cloud are compatible with the latest Windows 550 grid driver.
    

* Behavior Changes  
  * AKS now allows [daily schedules](https://learn.microsoft.com/azure/aks/planned-maintenance?tabs=azure-cli#schedule-types) for the auto upgrade configuration.
  * [Static Egress Gateway](https://learn.microsoft.com/azure/aks/configure-static-egress-gateway) memory limits increased from 500Mi to 3000Mi reducing the risk of memory-related restarts under load.
  * [HTTP Proxy](https://learn.microsoft.com/azure/aks/http-proxy?tabs=use-azure-cli) configuration updates will now trigger automatic reimaging with this release. This functionality was temporarily disabled.

>   * GPU-Provisioner is a component of [KAITO](https://learn.microsoft.com/azure/aks/ai-toolchain-operator), and it is now hidden from the customer as part of the KAITO add-on.

Reviewer 7: The GPU provisioner component of KAITO has now been moved to the AKS control plane when the KAITO add-on is used. The OSS installation will still require this component to run on the kubernetes nodes.

Author: updated.
